actors/builtin/miner/miner_actor.go

```diff
@@ -725,7 +736,7 @@ func (a Actor) ConfirmSectorProofsValid(rt Runtime, params *builtin.ConfirmSecto
 		}
 
 		// Unlock deposit for successful proofs, make it available for lock-up as initial pledge.
-		st.AddPreCommitDeposit(totalPledge.Neg())
+		st.AddPreCommitDeposit(totalPrecommitDeposit.Neg())
```

Comment:
I think this is the right way to do it but flagging here that we could alternatively compute IPs in the loop above and silently drop prove commits that won't have enough funds to process.  That would get tricky because unlocking gives you some funds, but we could apply decent heuristics (PD < IP for example) to figure out if including a prove commit risks failing all others.

Sending more than enough funds is in the miner's control and acceptable way to counter this risk.